src/zip_node.js

```diff
@@ -30,6 +45,42 @@ const zipNodeJs = async function ({ srcFiles, destPath, filename, mainFile, plug
   await endZip(archive, output)
 }
 
+// Zip a Node.js function file with esbuild
+const zipNodeJsWithEsbuild = async function ({
+  destFolder,
+  destPath,
+  externalModules = [],
+  filename,
+  mainFile,
+  pluginsModulesPath,
+}) {
+  const bundledFilePath = join(destFolder, filename)
+
+  await esbuild.build({
+    bundle: true,
+    entryPoints: [mainFile],
+    external: externalModules,
+    outfile: bundledFilePath,
+    platform: 'node',
+    target: ['es2017'],
```

Comment:
[sand] Should this be based on the [runtime version](https://docs.netlify.com/functions/build-with-javascript/#runtime-settings) of the user function?
We could get that value as an option.

Reply:
Absolutely. This would be an additional parameter which `@netlify/build` would need to pass, right?

Reply:
> Also we would like to configure esbuild to resolve from .netlify/plugins/node_modules, but again this can be done in another iteration (note that this is required for next/image to work).

@erezrokah can you add some more context to this one, please? 🙏🏻

Reply:
> Absolutely. This would be an additional parameter which `@netlify/build` would need to pass, right?

Yes I think so - pending the discussion on how to configure bundling.

> @erezrokah can you add some more context to this one, please? 🙏🏻

Of course! See https://github.com/netlify/zip-it-and-ship-it/pull/302 and https://github.com/netlify/pod-workflow/issues/116